### PR TITLE
[Merged by Bors] - feat: port Data.Complex.ExponentialBounds

### PR DESCRIPTION
**Pull request by Parcly-Taxel**
---

- [x] depends on: #4669

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

**Comment by semorrison** [comment]
This PR/issue depends on:
* leanprover-community/mathlib4#4669
By **[Dependent Issues](https://github.com/z0al/dependent-issues)** (🤖). Happy coding!

**Comment by semorrison** [comment]
This PR/issue depends on:
* ~~leanprover-community/mathlib4#4669~~
By **[Dependent Issues](https://github.com/z0al/dependent-issues)** (🤖). Happy coding!

File: Mathlib/Analysis/SpecialFunctions/Log/Deriv.lean [hunk 1]
```diff
@@ -0,0 +1,340 @@
...
+Porting note: TODO: use one of generic theorems about Taylor's series to prove this estimate.
+-/
+theorem abs_log_sub_add_sum_range_le {x : ℝ} (h : |x| < 1) (n : ℕ) :
+    |(∑ i in range n, x ^ (i + 1) / (i + 1)) + log (1 - x)| ≤ |x| ^ (n + 1) / (1 - |x|) := by
+  /- For the proof, we show that the derivative of the function to be estimated is small,
+    and then apply the mean value inequality. -/
+  let F : ℝ → ℝ := fun x => (∑ i in range n, x ^ (i + 1) / (i + 1)) + log (1 - x)
+  let F' : ℝ → ℝ := fun x ↦ -x ^ n / (1 - x)
+  -- Porting note: In `mathlib3`, the proof used `deriv`/`DifferentiableAt`. `simp` failed to
+  -- compute `deriv`, so I changed the proof to use `HasDerivAt` instead
```
Comment (semorrison) [inline Mathlib/Analysis/SpecialFunctions/Log/Deriv.lean:237]: Perhaps add a link to https://leanprover.zulipchat.com/#narrow/stream/144837-PR-reviews/topic/!4.234653.20Trigonometric.2EDeriv/near/363605590, where a related failure of `simp` to compute derivatives is being discussed.

Comment (semorrison) [reply Mathlib/Analysis/SpecialFunctions/Log/Deriv.lean:237]: Oh, too late, I see, that was the previous PR already. :-)

**Comment by ChrisHughes24** [comment]
bors merge

**Pull request closed without merging**